### PR TITLE
[move-prover][cli] Add benchmark functionality

### PR DESCRIPTION
**Pull request by wrwg**
## Description

This adds a new option `aptos move prove --benchmark` which lets verification run in the benchmark framework of the `move-prover/lab` tool. This tool verifies function by function and detects timeouts as well as measure verification time.

The result of the benchmark will be stored in `<move-package>/prover_benchmark.fun_data`. A graphic representation will be stored as well in `prover_benchmark.svg`.

The PR fixes also some other things on the way which came apparent when working on it, namely trying to identify the reasons for #15605:

- Adds new debug print for which verification targets are generated. This is reachable via `move prove -v debug`
- Reduces unnecessary verification of symmetric type instantiations `f<#1, #2>` and `f<#2, #1>`
- Adds an option `--skip-instance-check` to completely turn off verification of type instantiations.
- Removed legacy verification_analysis_v2 and global_invariant_instrumentation_v2. These are dead code since long  and confusing when trying to understand what functions are verified

## How Has This Been Tested?

Existing tests for not breaking something, and manually for the new feature

### CODE REVIEW

**Comment by wrwg** [comment]
* **#15636** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/15636?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/15636?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by wrwg** [comment]
Here is an example of generated SVG output file for an example from econia

prover_benchmark

**Review by rahxephon89 — APPROVED** [review]
LGTM, thanks!

File: third_party/move/move-prover/src/lib.rs [hunk 1]
```diff
@@ -71,19 +71,26 @@ pub fn run_move_prover_v2<W: WriteColor>(
...
     let compiler_options = move_compiler_v2::Options {
-        dependencies: cloned_options.move_deps,
-        named_address_mapping: cloned_options.move_named_address_values,
-        output_dir: cloned_options.output_path,
-        language_version: cloned_options.language_version,
+        dependencies: options.move_deps,
+        named_address_mapping: options.move_named_address_values,
+        output_dir: options.output_path,
+        language_version: options.language_version,
         compiler_version: None, // TODO: need to pass v2.x here
```
Comment (vineethk) [inline third_party/move/move-prover/src/lib.rs:87]: Just caught might eye (even if this is not related to this PR), we might want to use latest_stable compiler version here (or remove the TODO). Right now, it is probably being set based on the language version.

File: third_party/move/move-prover/lab/src/benchmark.rs [hunk 2]
```diff
@@ -149,27 +143,9 @@ fn run_benchmark(
...
-        }],
-        options.model_builder.clone(),
-        skip_attribute_checks,
-        &known_attributes,
-    )?;
-    let mut error_writer = StandardStream::stderr(ColorChoice::Auto);
+    options.skip_attribute_checks = true;
 
     if use_aptos_natives {
         options.backend.custom_natives =
```
Comment (vineethk) [inline third_party/move/move-prover/lab/src/benchmark.rs:151]: Again, not related directly to this PR, but it looks like we depend on non-third-party code here (`../../../../../aptos-move/framework/src/aptos-natives.bpl`) within third-party?

**Review by vineethk — APPROVED** [review]
